### PR TITLE
[SE-2725] Fixing GHA deprecated commands for deploy-storybook package

### PR DESCRIPTION
**Pull request by lorainegarutti**
[SE-2725] [GE-637]

### Description

- updating `docker/metadata-action` action version for `build-push-image package`
- updating internal packages for `build-push-release-image` and `deploy-storybook packages`
- updating `jenkins-job-trigger-action` and `actions/github-script` action version for `deploy-storybook` package
- updating `actions/github-script` action version for `extract-env-variables` package
- updating `google-github-actions/setup-gcloud` action version for `generate-gql-types` package

### How to test

- Tested first commit as internal reference 
- Tested second commit as external reference https://github.com/toptal/frontier-pub/pull/3970/commits/45d956b17c9b3076ca33c87e61685afa60887104

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/davinci-github-actions/blob/master/_docs/contribution/changeset-guidelines.md) (if needed)

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

[SE-2725]: https://toptal-core.atlassian.net/browse/SE-2725?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[GE-637]: https://toptal-core.atlassian.net/browse/GE-637?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: deploy-storybook/action.yml [hunk 1]
```diff
@@ -128,7 +128,7 @@ runs:
           }
         job_timeout: 1200
 
-    - uses: toptal/davinci-github-actions/extract-env-variables@v4.4.2
+    - uses: toptal/davinci-github-actions/extract-env-variables@ba935fc76996ca4aadc2a4999176a514b1cd5636
```
Comment (lorainegarutti) [inline deploy-storybook/action.yml:131]:
```suggestion
    - uses: toptal/davinci-github-actions/extract-env-variables@v4.8.3
```

**Comment by lorainegarutti** [comment]
@toptal-anvil ping reviewers

File: deploy-storybook/action.yml (hunk 1 above)
Comment (dmaklygin) [reply deploy-storybook/action.yml:131]: https://github.com/toptal/davinci-github-actions/pull/100
is already created. So, we can release these changes along together :)

**Review by dmaklygin — APPROVED** [review]
The update looks fine to me. I haven't tested it.

**Review by OndrejTuma — APPROVED** [review]
Looks good 👍 Thank you for contributing 🙇

**Review by stefanteixeira — APPROVED** [review]

**Pull request merged**